paprika/restraints/restraints.py

```diff
@@ -787,7 +785,19 @@ def static_DAT_restraint(
     rest = DAT_restraint()
     rest.continuous_apr = continuous_apr
     rest.amber_index = amber_index
-    rest.topology = pmd.load_file(ref_structure, structure=True)
+
+    if isinstance(ref_structure, pmd.amber._amberparm.AmberParm):
+        reference_trajectory = pt.load_parmed(ref_structure, traj=True)
+        rest.topology = ref_structure
+    elif isinstance(ref_structure, str):
+        reference_trajectory = pt.iterload(ref_structure, traj=True)
+        rest.topology = pmd.load_file(ref_structure, structure=True)
+    else:
+        raise Exception(
+            "static_DAT_restraint does not support the type associated with ref_structure:"
+            + type(ref_structure)
+        )
```

Comment:
It may be better to throw a more specific Exception here -- Maybe a `TypeError`?

Reply:
Thanks for changing this 👍